lib/remote-template.js

```diff
@@ -107,43 +107,45 @@ async function downloadTemplateElmJson(template, commit) {
   const {repoName, pathToFolder} = template;
   const pathToFolderAsUrl = pathToFolder ? `/${pathToFolder}` : '';
 
-  return got(
-    `https://raw.githubusercontent.com/${repoName}/${commit}${pathToFolderAsUrl}/elm.json`
-  )
-    .catch((error) => {
-      if (error.name === 'HTTPError' && error.response.statusCode === 404) {
-        const pathToElmJson = pathToFolder
-          ? `${pathToFolder.slice(1)}/elm.json`
-          : 'elm.json';
-
-        throw new ErrorMessage.CustomError(
-          // prettier-ignore
-          'TEMPLATE ELM.JSON NOT FOUND',
-          // prettier-ignore
-          `I found the ${chalk.yellow(repoName)} repository on GitHub,
+  let response;
+
+  try {
+    response = await got(
```

Comment:
This is one of the cases where I think it's easier to understand the flow of values when using `.then()` rather than async/await, mostly the fact that in the former there is no `let reponse;` pending without a value, and needed to understand/check that there is no code path where `response`'s value stays `undefined`.

`.catch()` being able to handle the errors until a certain point without having to split the code with a dangling `response` is part of the reason.

I think I'd rather extract functions to make it shorter if needed, rather than this change.

```js
return got(...)
  .catch(handleError)
  .then(handleResponse);

function handleResponse(response) {
  // ...
}
```

What do you think?

Reply:
Yeah, that's fair.